### PR TITLE
Add list localization #5280

### LINKED ISSUE #5280: Localization List Component
## Description  
Absent ability to localize List Component in the grid.
File path where describe mesages:
_projects/igniteui-angular/src/lib/list/list.component.html_ 

**_Absent localization for messages:_**
1. - There are no items in the list.
2. - Loading data from the server...

 * igniteui-angular version: ~7.3.1
 * browser: N/A

## Result  
Can't localized List Component in the grid. 

## Expected result  
Can localized List Component in the grid.

### PR DESCRIPTION
**Pull request by onlyexeption**
Closes #5280 

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/list/list.common.ts [hunk 1]
```diff
@@ -20,6 +22,15 @@ export class IgxListBaseDirective extends DisplayDensityBase {
...
     listItemRightPanningTemplate: IgxListItemRightPanningTemplateDirective;
+    private _resourceStrings = CurrentResourceStrings.ListResStrings;
+
+    set resourceStrings(value: IListResourceStrings) {
+        this._resourceStrings = Object.assign({}, this._resourceStrings, value);
+    }
+
+    get resourceStrings(): IListResourceStrings {
+        return this._resourceStrings;
+    }
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/list/list.common.ts:33]: You should move this to `IgxListComponent`. Also resourceStrings should be a public input and should have an API docs.

File: projects/igniteui-angular/src/lib/list/list.component.html [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
     <article class="message">
-        There are no items in the list.
+        {{this.resourceStrings.igx_list_no_items}}
     </article>
 </ng-template>
 
 <ng-template #defaultDataLoading>
     <article class="message">
-        Loading data from the server...
+        {{this.resourceStrings.igx_list_no_items}}
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/list/list.component.html:11]: It is not necessary to specify `this.` in the template.

Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/list/list.component.html]: This should be the loading string.

File: projects/igniteui-angular/src/lib/list/list.component.ts [hunk 3]
```diff
@@ -148,6 +150,23 @@ export class IgxListLineSubTitleDirective {
     providers: [{ provide: IgxListBaseDirective, useExisting: IgxListComponent }]
 })
 export class IgxListComponent extends IgxListBaseDirective {
+    /**
+    * An accessor that sets the resource strings.
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/list/list.component.ts:154]:
```suggestion
    * Sets the resource strings.
```

File: projects/igniteui-angular/src/lib/list/list.component.ts [hunk 4]
```diff
@@ -148,6 +150,23 @@ export class IgxListLineSubTitleDirective {
...
+    * By default it uses EN resources.
+    */
+   @Input()
+   set resourceStrings(value: IListResourceStrings) {
+       this._resourceStrings = Object.assign({}, this._resourceStrings, value);
+   }
+
+    /**
+    * An accessor that returns the resource strings.
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/list/list.component.ts:163]:
```suggestion
    * Returns the resource strings.
```

**Comment by onlyexeption** [comment]
Closing this PR as this feature will be included in versions >= 9.1.x

**Pull request closed without merging**